curp/tests/common/mod.rs

```diff
@@ -1,17 +1,59 @@
 #![allow(dead_code, unused)]
 
-use std::{env, io, time::Duration};
+use std::{
+    env, io,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
 
+use curp::role_change::RoleChange;
 use madsim::rand::{distributions::Alphanumeric, thread_rng, Rng};
+use parking_lot::RwLock;
 use thiserror::Error;
 use tracing_subscriber::fmt::time::{uptime, OffsetTime, Uptime};
+use utils::parking_lot_lock::RwLockMap;
 
 pub mod curp_group;
 pub mod test_cmd;
 
 pub const TEST_TABLE: &str = "test";
 pub const REVISION_TABLE: &str = "revision";
 
+#[derive(Default, Debug)]
+pub struct TestRoleChange {
+    pub inner: Arc<TestRoleChangeInner>,
+}
+
+#[derive(Default, Debug)]
+pub struct TestRoleChangeInner {
+    is_leader: AtomicBool,
+}
```

Comment:
Since the RawCurp depends on the `RoleChange` rather than `Arc<RoleChange>`.  Adding this inner struct is just for sharing it so that we can make sure those callbacks will be invoked at the right time and in the right place.